src/hooks/__tests__/useAlerts.test.ts

```diff
@@ -0,0 +1,83 @@
+import { act, renderHook } from '@testing-library/react';
+import { useAlerts } from '../useAlerts';
+
+describe('useAlerts', () => {
+  it('should add an alert and verify it is in the alerts list', () => {
+    const { result } = renderHook(() => useAlerts());
+
+    act(() => {
+      result.current.addAlert({ title: 'Test Alert', variant: 'success', id: 'alert-1' });
+    });
+
+    expect(result.current.alerts).toMatchSnapshot();
+  });
+
+  it('should add multiple alerts and verify all are present in the alerts list', () => {
+    const { result } = renderHook(() => useAlerts());
+    const alertCount = 3;
+
+    act(() => {
+      for (let i = 1; i <= alertCount; i++) {
+        result.current.addAlert({
+          id: `alert-${i}`,
+          title: `Test Alert ${i}`,
+          variant: i === 2 ? 'danger' : 'success'
+        });
+      }
+    });
+
+    expect(result.current.alerts).toMatchSnapshot();
+  });
+
+  it('should remove an alert by ID', () => {
+    const { result } = renderHook(() => useAlerts());
+
+    act(() => {
+      result.current.addAlert({ title: 'Test Alert', variant: 'success', id: 'alert-1' });
+    });
+
+    const addedAlertId = result.current.alerts[0].id;
+
+    act(() => {
+      result.current.removeAlert(addedAlertId);
+    });
+
+    expect(result.current.alerts).toHaveLength(0);
+  });
+
+  it('should remove an alert by title', () => {
+    const { result } = renderHook(() => useAlerts());
+    const targetAlertTitle = 'Test Alert 2';
+    const alertCount = 3;
+
+    act(() => {
+      for (let i = 1; i <= alertCount; i++) {
+        result.current.addAlert({
+          id: `alert-${i}`,
+          title: `Test Alert ${i}`,
+          variant: i === 2 ? 'danger' : 'success'
+        });
+      }
+    });
+
+    act(() => {
+      result.current.removeAlert(targetAlertTitle);
+    });
+
+    expect(result.current.alerts).toMatchSnapshot();
+  });
+
+  it('should handle removing an alert with a non-matching value gracefully', () => {
+    const { result } = renderHook(() => useAlerts());
+
+    act(() => {
+      result.current.addAlert({ title: 'Test Alert', variant: 'success' });
+    });
+
+    act(() => {
+      result.current.removeAlert('nonExistentValue');
+    });
+
+    expect(result.current.alerts).toHaveLength(1);
+  });
+});
```

Comment:
required: we can optimize the last 3 tests into a single `it`since technically we're doing the same thing just different properties. **Plus** we could include something additional, the ability to remove multiple alerts at once....

...using what you have, while we could use the returned "id" it feels more like a test if the list we put in really is the list we're getting out, let's make a mock instead...
```
  it('should allow removing alerts by any property', () => {
    const { result } = renderHook(() => useAlerts());

    // ...addAlertsHere

    act(() => {
      result.current.removeAlert(alerts[0].title); // leveraging the format like before
      result.current.removeAlert(alerts[1].id); // leveraging the format like before
      result.current.removeAlert('nonExistentValue'); // nonExistent like before
      result.current.removeAlert('danger'); // multiple alerts
    });

    // or you could do a snapshot, like expect(result.current.alerts).toMatchSnapshot('removed');
    expect(result.current.alerts).toHaveLength(1); 
   
  });
```